entity-framework/core/saving/transactions.md

```diff
@@ -111,81 +101,25 @@ public class BloggingContext : DbContext
 
-                    // Commit transaction if all commands succeed, transaction will auto-rollback
-                    // when disposed if either commands fails
-                    transaction.Commit();
-                }
-                catch (Exception)
-                {
-                    // TODO: Handle failure
-                }
-            }
-        }
-```
+[!code-csharp[Main](../../../samples/core/Saving/Saving/Transactions/ExternalDbTransaction/Sample.cs?name=Transaction&highlight=4,10,21,26,27,28)]
 
-## Using external DbTransactions (relational databases only)
+## Using System.Transactions
 
-If you are using multiple data access technologies to access a relational database, you may want to share a transaction between operations performed by these different technologies.
+> [!NOTE]  
+> This feature is new in EF Core 2.1.
 
-The following example, shows how to perform an ADO.NET SqlClient operation and an Entity Framework Core operation in the same transaction.
+It is possible to use ambient transactions if you need to coordinate across a larger scope.
 
-<!-- [!code-csharp[Main](samples/core/Saving/Saving/Transactions/ExternalDbTransaction/Sample.cs?highlight=4,10,21,26,27,28)] -->
-``` csharp
-        var connection = new SqlConnection(connectionString);
-        connection.Open();
+[!code-csharp[Main](../../../samples/core/Saving/Saving/Transactions/AmbientTransaction/Sample.cs?name=Transaction&highlight=1,24,25,26)]
 
-        using (var transaction = connection.BeginTransaction())
-        {
-            try
-            {
-                // Run raw ADO.NET command in the transaction
-                var command = connection.CreateCommand();
-                command.Transaction = transaction;
-                command.CommandText = "DELETE FROM dbo.Blogs";
-                command.ExecuteNonQuery();
-
-                // Run an EF Core command in the transaction
-                var options = new DbContextOptionsBuilder<BloggingContext>()
-                    .UseSqlServer(connection)
-                    .Options;
-
-                using (var context = new BloggingContext(options))
-                {
-                    context.Database.UseTransaction(transaction);
-                    context.Blogs.Add(new Blog { Url = "http://blogs.msdn.com/dotnet" });
-                    context.SaveChanges();
-                }
+It is also possible to enlist in an explicit transaction.
 
-                // Commit transaction if all commands succeed, transaction will auto-rollback
-                // when disposed if either commands fails
-                transaction.Commit();
-            }
-            catch (System.Exception)
-            {
-                // TODO: Handle failure
-            }
-```
+[!code-csharp[Main](../../../samples/core/Saving/Saving/Transactions/CommitableTransaction/Sample.cs?name=Transaction&highlight=1,13,26,27,28)]
```

Comment:
I would like to add a note explaining that as of version 2.1, .NET Core does not support using System.Transactions to coordinate transactions distributed across multiple resource managers. Sounds good?

Reply:
There are probably some other limitations, but I can't recall now.
For instance we could mention that non-relational providers might not support this.

Reply:
Fair enough. We should said that it is up to the provider in general.